### ISSUE TITLE
webdriver.remote.url interferes with AppiumDriver initialization (RemoteWebDriver returned)

### ISSUE THREAD

**Issue by maciejd** [serenity-bdd/serenity-core#411]
In my project, I execute tests on Appium node connected to Selenium Grid. The project also supports web tests.

I have (among others) the following properties defined:
```
driver = appium
appium.hub = {my grid url}
webdriver.remote.url = {my grid url}
```

It seems that if I have webdriver.remote.url defined, instead of an instance of AppiumDriver subclass, I'm getting a RemoteWebDriver. This blocks me from using a number of mobile-specific methods reserved for subclasses of AppiumDriver.

The workaround would be to always keep webdriver.remote.url empty when running Appium test suite. But this is a bit counter-intuitive.

If you also find this being an issue (and not a feature), then WebDriverFactory is probably to blame. Once shouldUseARemoteDriver() returns true, it goes straight for RemoteWebDriver, not even bothering about isAnAppiumDriver().

```
protected synchronized WebDriver newWebdriverInstance(final Class<? extends WebDriver> driverClass) {
        try {
            WebDriver driver;
            if (isARemoteDriver(driverClass) || shouldUseARemoteDriver() || saucelabsUrlIsDefined() || browserStackUrlIsDefined()) {
                driver = newRemoteDriver();
            } else if (isAnAppiumDriver(driverClass)) {
                driver = appiumDriver();
```

**Comment by wakaleo** [comment]
Hi @maciejd,

Thanks for this issue. This should be fixed in the next release.

**Comment by wakaleo** [comment]
Fix released in version 1.1.32-rc.2
